app.py

```diff
@@ -0,0 +1,168 @@
+        if 'polyglot' in DOWNLOADED_MODELS and lan in DOWNLOADED_MODELS['polyglot']:
+            model = 'polyglot'
+            ents = get_poly_ents(text, lan)
+            return jsonify({'language': lan, 'model': model, 'entities': ents}), 200
+        else:
+            return jsonify({'error': f'no polyglot model for language {lan}'}), 500
+    else:
+        if 'spacy' in DOWNLOADED_MODELS and model in DOWNLOADED_MODELS['spacy']:
+            nlp = spacy.load(model)
+            ents = get_spacy_ents(text, nlp)
+            return jsonify({'language': lan, 'model': model, 'entities': ents}), 200
+        else:
+            return jsonify({'error': f'no spacy model with name {model}'}), 500
+
+
+@app.route('/entity_extraction', methods=['POST'])
+def get_entities():
+    """Entity extraction endpoint.
+    It checks whether it has received a text data and whether a language or an
+    actual model was specified by the request. If not, the language is
+    determined and the most appropriate model which is installed is used in
+    order to extract entities. Spacy models have prioriy over polyglot models.
+    larger spacy models have priority over smalles ones for the same language.
+    Spacy's specific language models have priority over the multilingual model.
+    A model can also be specified during the request. Either, a spacy model name
+    has to be provided, or NER via polyglot with a language by adding
+    - "model": "poly_{language_code}" to the request.
+    It is first checked, whether the model is installed, if not, the server
+    will respond with a 500 error.
+    """
+    if not request.json or 'text' not in request.json:
+        abort(400)
+    text = request.json['text']
+    if 'lan' in request.json:
+        lan = request.json['lan']
+    else:
+        lan = get_language(text[:2500])
+    if 'model' in request.json:
+        model = request.json['model']
+        response = get_ents_from_model(text, lan, model)
+    elif lan in SPACY_LANGUAGES:
+        model, nlp = get_spacy_model(lan)
+        ents = get_spacy_ents(text, nlp)
+        response = jsonify({'language': lan, 'model': model, 'entities': ents}), 200
+    elif lan in DOWNLOADED_MODELS['polyglot']:
+        model = 'polyglot'
+        ents = get_poly_ents(text, lan)
+        response = jsonify({'language': lan, 'model': model, 'entities': ents}), 200
+    else:
+        response = jsonify({'error': f'no model for text in language {lan}'}), 500
+    return response
+
+
+@app.route('/language_detection', methods=['POST'])
+def return_language():
+    """ Language Detection endpoint. Gets the language based on the text and
+    returns it in a response.
+    """
+    if not request.json or 'text' not in request.json:
+        abort(400)
+    language = get_language(request.json['text'])
+    if language:
+        return jsonify({'language': language}), 200
+    else:
+        return jsonify({'error': 'could not determine language for text'}, 500)
+
```

Comment:
Add another route `GET /config` to this service that returns current configuration. We will use this one as a health check in nomad/consul.